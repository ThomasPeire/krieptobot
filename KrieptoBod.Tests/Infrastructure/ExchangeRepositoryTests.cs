using KrieptoBod.Infrastructure.Exchange;
using KrieptoBod.Model;
using Moq;
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace KrieptoBod.Tests.Infrastructure
{
    public class Tests
    {
        private readonly Mock<IExchangeService> _exchangeService = new Mock<IExchangeService>();

        [SetUp]
        public void Setup()
        {
            //Todo use mock data
            _exchangeService
                .Setup(service => service.GetBalanceAsync())
                .Returns(Task.FromResult(new List<Balance>().AsEnumerable()));
            _exchangeService
                .Setup(service => service.GetCandlesAsync(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<int>(), It.IsAny<DateTime>(), It.IsAny<DateTime>()))
                .Returns(Task.FromResult(new List<Candle>().AsEnumerable()));

            _exchangeService
                .Setup(service => service.GetMarketsAsync())
                .Returns(Task.FromResult(new List<Market>().AsEnumerable()));

            _exchangeService
                .Setup(service => service.GetMarketAsync(It.IsAny<string>()))
                .Returns(Task.FromResult(new Market()));

            _exchangeService
                .Setup(service => service.GetAssetsAsync())
                .Returns(Task.FromResult(new List<Asset>().AsEnumerable()));

            _exchangeService
                .Setup(service => service.GetAssetAsync(It.IsAny<string>()))
                .Returns(Task.FromResult(new Asset()));

            _exchangeService
                .Setup(service => service.GetTradesAsync(It.IsAny<string>(), It.IsAny<int>(), It.IsAny<DateTime>(), It.IsAny<DateTime>(), It.IsAny<Guid>(), It.IsAny<Guid>()))
                .Returns(Task.FromResult(new List<Trade>().AsEnumerable()));

            _exchangeService
                .Setup(service => service.GetOrderAsync(It.IsAny<string>(), It.IsAny<Guid>()))
                .Returns(Task.FromResult(new Order()));

            _exchangeService
                .Setup(service => service.GetOrdersAsync(It.IsAny<string>(), It.IsAny<int>(), It.IsAny<DateTime>(), It.IsAny<DateTime>(), It.IsAny<Guid>(), It.IsAny<Guid>()))
                .Returns(Task.FromResult(new List<Order>().AsEnumerable()));

            _exchangeService
                .Setup(service => service.GetOpenOrderAsync(It.IsAny<string>()))
                .Returns(Task.FromResult(new Order()));

        }

        [Test]
        public async Task GetBalance_ShouldReturn_Balance()
        {
            var exchangeRepository = new ExchangeRepository(_exchangeService.Object);

            var result = await exchangeRepository.GetBalanceAsync();

            Assert.That(result, Is.Not.Null);
        }

        [Test]
        public async Task GetCandles_ShouldReturn_Candles()
        {
            var exchangeRepository = new ExchangeRepository(_exchangeService.Object);

            var result = await exchangeRepository.GetCandlesAsync("", "", 0, DateTime.Now, DateTime.Now);

            Assert.That(result, Is.Not.Null);
        }

        [Test]
        public void GetMarkets_ShouldThrow_NotImplementedException()
        {
            Assert.That(async () => await new ExchangeRepository(_exchangeService.Object).GetMarketsAsync(), Throws.Exception);
        }

        [Test]
        public void GetMarket_ShouldThrow_NotImplementedException()
        {
            Assert.That(async () => await new ExchangeRepository(_exchangeService.Object).GetMarketAsync(""), Throws.Exception);
        }

        [Test]
        public void GetAssets_ShouldThrow_NotImplementedException()
        {
            Assert.That(async() => await new ExchangeRepository(_exchangeService.Object).GetAssetsAsync(), Throws.Exception);
        }

        [Test]
        public void GetAsset_ShouldThrow_NotImplementedException()
        {
            Assert.That(async () => await new ExchangeRepository(_exchangeService.Object).GetAssetAsync(""), Throws.Exception);
        }

        [Test]
        public void GetTrades_ShouldThrow_NotImplementedException()
        {
            Assert.That(async () => await new ExchangeRepository(_exchangeService.Object).GetTradesAsync(""), Throws.Exception);
        }

        [Test]
        public void GetOrders_ShouldThrow_NotImplementedException()
        {
            Assert.That(async () => await new ExchangeRepository(_exchangeService.Object).GetOrdersAsync(""), Throws.Exception);
        }

        [Test]
        public void GetOrder_ShouldThrow_NotImplementedException()
        {
            Assert.That(async () => await new ExchangeRepository(_exchangeService.Object).GetOrderAsync("", new Guid()), Throws.Exception);
        }

        [Test]
        public void GetOpenOrder_ShouldThrow_NotImplementedException()
        {
            Assert.That(async () => await new ExchangeRepository(_exchangeService.Object).GetOpenOrderAsync(""), Throws.Exception);
        }
    }
}