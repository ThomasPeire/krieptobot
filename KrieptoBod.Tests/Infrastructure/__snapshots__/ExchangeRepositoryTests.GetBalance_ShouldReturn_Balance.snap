[
  {
    "Symbol": "EUR",
    "Available": 0.0,
    "InOrder": 1103.0
  },
  {
    "Symbol": "ADA",
    "Available": 0.0,
    "InOrder": 0.0
  },
  {
    "Symbol": "ARK",
    "Available": 0.0,
    "InOrder": 0.0
  },
  {
    "Symbol": "BAT",
    "Available": 0.0,
    "InOrder": 0.0
  },
  {
    "Symbol": "BTC",
    "Available": 24.0,
    "InOrder": 0.0
  },
  {
    "Symbol": "CHZ",
    "Available": 0.0,
    "InOrder": 0.0
  },
  {
    "Symbol": "CMT",
    "Available": 0.0,
    "InOrder": 0.0
  },
  {
    "Symbol": "ETH",
    "Available": 1959154.0,
    "InOrder": 0.0
  },
  {
    "Symbol": "GAS",
    "Available": 0.0,
    "InOrder": 0.0
  },
  {
    "Symbol": "HOT",
    "Available": 0.0,
    "InOrder": 0.0
  },
  {
    "Symbol": "LTC",
    "Available": 11392.0,
    "InOrder": 0.0
  },
  {
    "Symbol": "NPXS",
    "Available": 0.0,
    "InOrder": 0.0
  },
  {
    "Symbol": "TRX",
    "Available": 0.0,
    "InOrder": 0.0
  },
  {
    "Symbol": "XEM",
    "Available": 0.0,
    "InOrder": 0.0
  }
]
